action.yml

```diff
@@ -39,4 +52,17 @@ runs:
           -v "${{ inputs.terraform-var-file }}" \
           -p "${{ inputs.terraform-plan-file }}" \
           -d "${{ inputs.destroy-mode }}"
+      id: tf_plan
+      shell: bash
+    - run: |
+        ${{ github.action_path }}/tf-show.sh \
+          -r ${{ inputs.region }} \
+          -a ${{ inputs.access-key }} \
+          -s ${{ inputs.secret-key }} \
+          -t ${{ inputs.terraform-folder }} \
+          -b ${{ inputs.backend-configuration }} \
+          -p ${{ inputs.terraform-plan-file }} >> tf_plan.txt
+        ${{ github.action_path }}/add-iac-change-to-job-summary.sh tf_plan.txt
+      id: add_iac_changes_to_summary
+      if: ${{ steps.tf_plan.outputs.tf_detect_changes == 'true' }}
```

Comment:
the previous if is the key, this step (add iac change to job summary) will only happen if the tf plan detect changes